### PR TITLE
feat(addon-table): add paginationChange event

### LINKED ISSUE #2044: 🐞 - table-pagination - sizeChange souldn't call pageChange
### Which @taiga-ui/* package(s) are the source of the bug?

addon-table

### Please provide a link to a minimal reproduction of the bug

https://stackblitz.com/edit/taiga-starter

### Is this issue blocking you?

Blocking

### Description

Hi there, at first, thank you for amazing UI Library. It's the greatest!
A few days ago I decided to use TaigaUI for my project and met a proplem. 

In my application router query params and pagination are synced, so query string is a source of truth for the pagination. 
It looks like when a user change something on the page (pageIndex, perPage, etc), at first, I use router to change the query string, and then, state from query string changes the pagination on a page. 
Unfortunately, it doesn't work well, because when I change page size, table-pagination unexpectedly calls pageChange event and router.navigate calls twice (For page size changing and then for current page changing). But angular can't handle it (two navigates an once) and I got an error
`router.mjs:5931 Router Event: NavigationCancel`
`router.mjs:5932 NavigationCancel(id: 6, url: '/crud?page=1&per_page=20')`
`router.mjs:5933 NavigationCancel {id: 6, url: '/crud?page=1&per_page=20', reason: 'Navigation ID 6 is not equal to the current navigation id 7'}`
It means that angular can't update query params, because another navigation has started.

I think it heppens [here](https://github.com/Tinkoff/taiga-ui/blob/main/projects/addon-table/components/table-pagination/table-pagination.component.ts#L77). As we can see thera are two call at once.

`this.sizeChange.emit`
`this.pageChange.emit(this.page)`

In my opinion this behavior is not trasparent and I'd like to have an ability to change the page on my own with page size (at once) to prevent this error.

Thank you!

### Angular version

13.2.0

### Taiga UI version

2.50.0

### Which browsers have you used?

- [X] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### Which operating systems have you used?

- [ ] macOS
- [X] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

### PR DESCRIPTION
**Pull request by xvs32x**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [x] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

Closes #2044 

## What is the new behavior?
This PR adding an opportunity to subscribe to both of parameters (page and size) at once and let us to prevent behavior from the issue

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by xvs32x** [comment]
Hi, @splincode. What do you think?

**Review by splincode — APPROVED** [review]

File: projects/addon-table/components/table-pagination/table-pagination.component.ts [hunk 1]
```diff
@@ -37,6 +38,9 @@ export class TuiTablePaginationComponent {
     @Output()
     readonly sizeChange = new EventEmitter<number>();
 
+    @Output()
+    readonly paginationChange = new EventEmitter<TuiPagination>();
```
Comment (nsbarsukov) [inline projects/addon-table/components/table-pagination/table-pagination.component.ts:42]: Update API page, please:

https://taiga-ui.dev/components/table-pagination/API

https://github.com/Tinkoff/taiga-ui/blob/main/projects/demo/src/modules/tables/table-pagination/table-pagination.template.html

File: projects/addon-table/components/table-pagination/table-pagination.component.ts [hunk 2]
```diff
@@ -66,6 +70,13 @@ export class TuiTablePaginationComponent {
         return this.end === this.total;
     }
 
+    get pagination(): TuiPagination {
+        return {
+            page: this.page,
+            size: this.size,
```
Comment (nsbarsukov) [inline projects/addon-table/components/table-pagination/table-pagination.component.ts:76]: Run prettier for this file

File: projects/addon-table/components/table-pagination/table-pagination.component.ts (hunk 1 above)
Comment (xvs32x) [reply projects/addon-table/components/table-pagination/table-pagination.component.ts:42]: done

File: projects/addon-table/components/table-pagination/table-pagination.component.ts (hunk 2 above)
Comment (xvs32x) [reply projects/addon-table/components/table-pagination/table-pagination.component.ts:76]: done

**Comment by nsbarsukov** [comment]
[Lint](https://github.com/Tinkoff/taiga-ui/actions/runs/3575585008/jobs/6023726505)-job fails with error:

```
Run npm run prettier -- --check

> @taiga-ui/components@3.11.0 prettier
> prettier '**/*.{svg,yml,js,ts,html,md,less,json}' --cache node_modules/.cache/prettier --check

Checking formatting...
[warn] projects/demo/src/modules/tables/table-pagination/table-pagination.template.html
[warn] Code style issues found in the above file. Forgot to run Prettier?
```

**Comment by waterplea** [comment]
I think we should also deprecate original outputs since this is the preferred way now, right?

**Comment by xvs32x** [comment]
> I think we should also deprecate original outputs since this is the preferred way now, right?

Yeah, it makes sense. Could you add some examples, how i can deprecate the props?

**Comment by waterplea** [comment]
> > I think we should also deprecate original outputs since this is the preferred way now, right?
> 
> Yeah, it makes sense. Could you add some examples, how I can deprecate the props?

Just write:
```ts
/**
 * TODO: Remove in 4.0
 * @deprecated use paginationChange
 */
```
over those outputs

File: projects/addon-table/components/table-pagination/pagination.model.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+export interface TuiPagination {
```
Comment (splincode) [inline projects/addon-table/components/table-pagination/pagination.model.ts:1]: `pagination.model.ts` -> `table-pagination.ts`

File: projects/addon-table/components/table-pagination/table-pagination.component.ts [hunk 4]
```diff
@@ -4,6 +4,7 @@ import {tuiDefaultProp} from '@taiga-ui/cdk';
 import {TUI_SPIN_TEXTS} from '@taiga-ui/core';
 import {Observable} from 'rxjs';
 
+import {TuiPagination} from './pagination.model';
```
Comment (splincode) [inline projects/addon-table/components/table-pagination/table-pagination.component.ts:7]: `TuiTablePagination`

File: projects/demo/src/modules/tables/table-pagination/table-pagination.template.html [hunk 5]
```diff
@@ -123,6 +123,19 @@
                 <code>size</code>
                 -property (user selects new amount of lines per page).
             </ng-template>
+            <ng-template
+                #documentationPropertyPaginationChange="documentationProperty"
+                i18n
+                documentationPropertyName="paginationChange"
+                documentationPropertyMode="output"
+                documentationPropertyType="TuiPagination"
```
Comment (splincode) [inline projects/demo/src/modules/tables/table-pagination/table-pagination.template.html:131]: `TuiTablePagination`

Comment (xvs32x) [reply projects/demo/src/modules/tables/table-pagination/table-pagination.template.html]: done

**Pull request merged**